File: app/src/main/java/com/eam/investire/activity/CadastroActivity.java
package com.eam.investire.activity;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.eam.investire.R;
import com.eam.investire.config.ConfiguracaoFirebase;
import com.eam.investire.helper.Base64Custom;
import com.eam.investire.model.Usuario;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseAuthInvalidCredentialsException;
import com.google.firebase.auth.FirebaseAuthUserCollisionException;
import com.google.firebase.auth.FirebaseAuthWeakPasswordException;

import java.util.Objects;

public class CadastroActivity extends AppCompatActivity {

    private EditText campoNome, campoEmail, campoSenha;
    private Usuario usuario;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_cadastro);

        campoNome = findViewById(R.id.editNome);
        campoEmail = findViewById(R.id.editEmail);
        campoSenha = findViewById(R.id.editSenha);
        Button botaoCadastrar = findViewById(R.id.buttonCadastrar);

        botaoCadastrar.setOnClickListener(v -> {
            String textoNome = campoNome.getText().toString();
            String textoEmail = campoEmail.getText().toString();
            String textoSenha = campoSenha.getText().toString();

            if (!textoNome.isEmpty()) {
                if (!textoEmail.isEmpty()) {
                    if (!textoSenha.isEmpty()) {
                        usuario = new Usuario();
                        usuario.setNome(textoNome);
                        usuario.setEmail(textoEmail);
                        usuario.setSenha(textoSenha);
                        cadastrarUsuario();
                    } else {
                        Toast.makeText(CadastroActivity.this,
                                "Preencha a senha!",
                                Toast.LENGTH_SHORT).show();
                    }
                } else {
                    Toast.makeText(CadastroActivity.this,
                            "Preencha o e-mail!",
                            Toast.LENGTH_SHORT).show();
                }
            } else {
                Toast.makeText(CadastroActivity.this,
                        "Preencha o nome!",
                        Toast.LENGTH_SHORT).show();
            }
        });
    }

    public void cadastrarUsuario() {
        FirebaseAuth autenticacao = ConfiguracaoFirebase.getFirebaseAutenticacao();
        autenticacao.createUserWithEmailAndPassword(
                usuario.getEmail(),
                usuario.getSenha()
        ).addOnCompleteListener(this, task -> {
            if (task.isSuccessful()) {
                String idUsuario = Base64Custom.codificarBase64(usuario.getEmail());
                usuario.setIdUsuario(idUsuario);
                usuario.salvar();
                finish();
            } else {
                String excecao;
                try {
                    throw Objects.requireNonNull(task.getException());
                } catch (FirebaseAuthWeakPasswordException e) {
                    excecao = "Digite uma senha mais forte!";
                } catch (FirebaseAuthInvalidCredentialsException e) {
                    excecao = "Digite um e-mail válido!";
                } catch (FirebaseAuthUserCollisionException e) {
                    excecao = "E-mail já cadastrado!";
                } catch (Exception e) {
                    excecao = "Erro ao cadastrar usuário" + e.getMessage();
                    e.printStackTrace();
                }

                Toast.makeText(CadastroActivity.this,
                        excecao,
                        Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: app/src/main/java/com/eam/investire/activity/CadastroActivity.kt
package com.eam.investire.activity

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.eam.investire.R
import com.eam.investire.config.ConfiguracaoFirebase
import com.eam.investire.helper.Base64Custom
import com.eam.investire.model.Usuario
import com.google.android.gms.tasks.Task
import com.google.firebase.auth.AuthResult
import com.google.firebase.auth.FirebaseAuthInvalidCredentialsException
import com.google.firebase.auth.FirebaseAuthUserCollisionException
import com.google.firebase.auth.FirebaseAuthWeakPasswordException
import java.util.*

class CadastroActivity : AppCompatActivity() {
    private var campoNome: EditText? = null
    private var campoEmail: EditText? = null
    private var campoSenha: EditText? = null
    private var usuario: Usuario? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_cadastro)
        campoNome = findViewById(R.id.editNome)
        campoEmail = findViewById(R.id.editEmail)
        campoSenha = findViewById(R.id.editSenha)
        val botaoCadastrar = findViewById<Button>(R.id.buttonCadastrar)
        botaoCadastrar.setOnClickListener {
            val textoNome = campoNome!!.text.toString()
            val textoEmail = campoEmail!!.text.toString()
            val textoSenha = campoSenha!!.text.toString()
            if (textoNome.isNotEmpty()) {
                if (textoEmail.isNotEmpty()) {
                    if (textoSenha.isNotEmpty()) {
                        usuario = Usuario()
                        usuario!!.nome = textoNome
                        usuario!!.email = textoEmail
                        usuario!!.senha = textoSenha
                        cadastrarUsuario()
                    } else {
                        Toast.makeText(this@CadastroActivity,
                                "Preencha a senha!",
                                Toast.LENGTH_SHORT).show()
                    }
                } else {
                    Toast.makeText(this@CadastroActivity,
                            "Preencha o e-mail!",
                            Toast.LENGTH_SHORT).show()
                }
            } else {
                Toast.makeText(this@CadastroActivity,
                        "Preencha o nome!",
                        Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun cadastrarUsuario() {
        val autenticacao = ConfiguracaoFirebase.firebaseAutenticacao
        usuario!!.email?.let {
            usuario!!.senha?.let { it1 ->
                autenticacao!!.createUserWithEmailAndPassword(
                        it,
                        it1
            ).addOnCompleteListener(this) { task: Task<AuthResult?> ->
                if (task.isSuccessful) {
                    val idUsuario = usuario!!.email?.let { it2 -> Base64Custom.codificarBase64(it2) }
                    usuario!!.setIdUsuario(idUsuario)
                    usuario!!.salvar()
                    finish()
                } else {
                    val excecao: String
                    try {
                        throw Objects.requireNonNull(task.exception)
                                ?: throw NullPointerException("Expression 'Objects.requireNonNull(task.exception)' must not be null")
                    } catch (e: FirebaseAuthWeakPasswordException) {
                        excecao = "Digite uma senha mais forte!"
                    } catch (e: FirebaseAuthInvalidCredentialsException) {
                        excecao = "Digite um e-mail válido!"
                    } catch (e: FirebaseAuthUserCollisionException) {
                        excecao = "E-mail já cadastrado!"
                    } catch (e: Exception) {
                        excecao = "Erro ao cadastrar usuário" + e.message
                        e.printStackTrace()
                    }
                    Toast.makeText(this@CadastroActivity,
                            excecao,
                            Toast.LENGTH_SHORT).show()
                }
            }
            }
        }
    }
}